functions/snippets/src/main/java/com/example/functions/OcrProcessImage.java

```diff
@@ -0,0 +1,161 @@
+
+    ImageSource imgSource = ImageSource.newBuilder().setGcsImageUri(gcsPath).build();
+    Image img = Image.newBuilder().setSource(imgSource).build();
+
+    Feature textFeature = Feature.newBuilder().setType(Feature.Type.TEXT_DETECTION).build();
+    AnnotateImageRequest visionRequest =
+        AnnotateImageRequest.newBuilder().addFeatures(textFeature).setImage(img).build();
+    visionRequests.add(visionRequest);
+
+    // Detect text in an image using the Cloud Vision API
+    AnnotateImageResponse visionResponse;
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
+      visionResponse = client.batchAnnotateImages(visionRequests).getResponses(0);
+      if (visionResponse == null || !visionResponse.hasFullTextAnnotation()) {
+        LOGGER.info(String.format("Image %s contains no text", filename));
+        return;
+      }
+
+      if (visionResponse.hasError()) {
+        throw new RuntimeException(
+            "Error detecting text: " + visionResponse.getError().getMessage());
+      }
+    } catch (IOException e) {
+      // Cast to RuntimeException
+      throw new RuntimeException(e);
+    }
+
+    String text = visionResponse.getFullTextAnnotation().getText();
+    LOGGER.info("Extracted text from image: " + text);
+
+    // Detect language using the Cloud Translation API
+    DetectLanguageRequest languageRequest =
+        DetectLanguageRequest.newBuilder()
+            .setParent(LOCATION_NAME.toString())
+            .setMimeType("text/plain")
+            .setContent(text)
+            .build();
+    DetectLanguageResponse languageResponse;
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
+      languageResponse = client.detectLanguage(languageRequest);
+    } catch (IOException e) {
+      // Log error
+      LOGGER.severe("Error detecting language: " + e.getMessage());
+      e.printStackTrace();
+      return;
+    }
+
+    if (languageResponse.getLanguagesCount() == 0) {
+      LOGGER.info("No languages were detected for text: " + text);
+      return;
+    }
+
+    String languageCode = languageResponse.getLanguages(0).getLanguageCode();
+    LOGGER.info(String.format("Detected language %s for file %s", languageCode, filename));
+
+    // Send a Pub/Sub translation request for every language we're going to translate to
+    for (String targetLanguage : TO_LANGS) {
+      LOGGER.info("Sending translation request for language " + targetLanguage);
+      OcrTranslateApiMessage message = new OcrTranslateApiMessage(text, filename, targetLanguage);
+      ByteString byteStr = ByteString.copyFrom(message.toPubsubData());
+      PubsubMessage pubsubApiMessage = PubsubMessage.newBuilder().setData(byteStr).build();
+      try {
+        publisher.publish(pubsubApiMessage).get();
+      } catch (InterruptedException | ExecutionException e) {
+        // Log error
+        LOGGER.severe("Error publishing translation request: " + e.getMessage());
+        e.printStackTrace();
```

Comment:
printStackTrace only goes to stdout
```suggestion
        LOGGER.log(Level.SEVERE, "Error publishing translation request: " + e.getMessage(), e);
```

Reply:
(please fix everywhere else)

Reply:
Done.